operator/controllers/csiscaleoperator_controller.go

```
@@ -219,6 +221,72 @@ func (r *CSIScaleOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
+	configMap := &corev1.ConfigMap{}
+	cerr := r.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      config.CSIConfigMap,
+		Namespace: req.Namespace,
+	}, configMap)
+	if cerr != nil {
+		if !errors.IsNotFound(cerr) {
+			message := "Failed to get ConfigMap resource " + config.CSIConfigMap
+			logger.Error(cerr, message)
+			// TODO: Add event.
+			meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
+				Type:    string(config.StatusConditionSuccess),
+				Status:  metav1.ConditionFalse,
+				Reason:  string(csiv1.ResourceReadError),
+				Message: message,
+			})
+			return ctrl.Result{}, cerr
+		}
+	} else {
+		clustersBytes, err := json.Marshal(&instance.Spec.Clusters)
+		if err != nil {
+			logger.Error(err, "Failed to marshal clusters data of this instance")
+			return ctrl.Result{}, err
+		}
+		clustersString := string(clustersBytes)
+		configMapDataBytes, err := json.Marshal(&configMap.Data)
+		if err != nil {
+			logger.Error(err, "Failed to marshal data of ConfigMap "+config.CSIConfigMap)
+			return ctrl.Result{}, err
+		}
+		configMapDataString := string(configMapDataBytes)
+		configMapDataString = strings.Replace(configMapDataString, " ", "", -1)
+		configMapDataString = strings.Replace(configMapDataString, "\\\"", "\"", -1)
+
+		if !strings.Contains(configMapDataString, clustersString) {
+			logger.Info("Some of the cluster fields of CSIScaleOperator instance are changed, so restarting node plugin pods")
+			daemonSet, derr := r.getNodeDaemonSet(instance)
+			if derr != nil {
+				if !errors.IsNotFound(derr) {
+					message := "Failed to get node plugin Daemonset"
+					logger.Error(derr, message)
+					// TODO: Add event.
+					meta.SetStatusCondition(&crStatus.Conditions, metav1.Condition{
+						Type:    string(config.StatusConditionSuccess),
+						Status:  metav1.ConditionFalse,
+						Reason:  string(csiv1.ResourceReadError),
+						Message: message,
+					})
+					return ctrl.Result{}, derr
+				}
```

If derr != nil and if "if !errors.IsNotFound(derr) {" fails then we should set status and return after this line, right ?

---

If derr is not nil, that means we have got some error here, and if that error is `IsNotFound` - we can ignore this error as the daemonset won't be there for the first time reconciliation. So setting status and restarting reconciliation (by returning error) only in case of err is not nil and that non-nil error is not `IsNotFound`.

---

Ok, got it.